src/open_clip/hf_model.py

```diff
@@ -239,3 +240,131 @@ def set_grad_checkpointing(self, enable=True):
 
     def init_parameters(self):
         pass
+
+
+
+class HFVisionEncoder(nn.Module):
+    """HuggingFace model adapter"""
+
+    output_tokens: torch.jit.Final[bool]
+
+    def __init__(
+        self,
+        model_name_or_path: str,
+        image_size: int,
+        output_dim: int,
+        config: PretrainedConfig = None,
+        pool_type: str = 'tok',
+        proj_type: Optional[str] = None,
+        pretrained: bool = True,
+        output_tokens: bool = False,
+        trust_remote_code: bool = False,
+        revision: Optional[str] = None,
+    ):
+        super().__init__()
+        self.output_tokens = output_tokens
+        self.output_dim = output_dim
+        self.image_size = to_2tuple(image_size)
+
+        if transformers is None:
+            raise RuntimeError(
+                'Please `pip install transformers` to use pre-trained HuggingFace models'
+            )
+        if config is None:
+            self.config = AutoConfig.from_pretrained(
+                model_name_or_path,
+                trust_remote_code=trust_remote_code,
+                code_revision=revision,
```

Comment:
is `code_revision` a valid signature? i thoguht it is `revision`?

Reply:
I think they changed it

Reply:
I removed revision from vision model anyway. If needed we can change it for text encoder